src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1277,16 +1278,33 @@ public void updateApplicationStaging(String applicationName, Staging staging) {
     }
 
     @Override
-    public void updateApplicationUris(String applicationName, List<String> uris) {
+    public void updateApplicationRoutes(String applicationName, Set<CloudRouteSummary> routes) {
         CloudApplication application = getApplication(applicationName);
-        List<String> newUris = new ArrayList<>(uris);
-        newUris.removeAll(application.getUris());
-        List<String> removeUris = new ArrayList<>(application.getUris());
-        removeUris.removeAll(uris);
-        removeUris(removeUris, application.getMetadata()
-                                          .getGuid());
-        addUris(newUris, application.getMetadata()
-                                    .getGuid());
+        Set<CloudRouteSummary> currentRoutes = application.getRoutes();
+
+        Set<CloudRouteSummary> outdatedRoutes = selectRoutesOnlyPresentInFirst(currentRoutes, routes);
+
+        Set<CloudRouteSummary> newRoutes = selectRoutesOnlyPresentInFirst(routes, currentRoutes);
+
+        removeRoutes(outdatedRoutes, application.getMetadata()
+                                                .getGuid());
+        addRoutes(newRoutes, application.getMetadata()
+                                        .getGuid());
+    }
+
+    protected Set<CloudRouteSummary> selectRoutesOnlyPresentInFirst(Set<CloudRouteSummary> allRoutes,
+                                                                  Set<CloudRouteSummary> routesToSubtract) {
+        Set<CloudRouteSummary> selectedRoutes = new HashSet<>();
```

Comment:
Why note adding equals() in CloudRouteSummary and just use: allRoutes.removeAll(routesToSubtract)
In this we can remove this function at all